conjure-core/src/test/java/com/palantir/conjure/defs/validator/ConjureSourceFileValidatorTest.java

```diff
@@ -511,6 +511,28 @@ public void testSafetyMarkersAreNotAllowedWhenSafetyIsRequired() {
                 .hasMessageContaining("Service.end(arg): Safety markers have been replaced by the 'safety' field");
     }
 
+    @Test
+    public void testSafetyOnExternalImport() {
+        ExternalReference external = ExternalReference.builder()
+                .safety(LogSafety.DO_NOT_LOG)
+                .externalReference(TypeName.of("Long", "java.lang"))
+                .fallback(Type.primitive(PrimitiveType.BOOLEAN))
+                .build();
+        ConjureDefinition conjureDef = ConjureDefinition.builder()
+                .version(1)
+                .types(TypeDefinition.object(ObjectDefinition.builder()
+                        .typeName(FOO)
+                        .fields(FieldDefinition.builder()
+                                .fieldName(FieldName.of("bad"))
+                                .type(Type.external(external))
+                                .safety(LogSafety.UNSAFE)
+                                .docs(DOCS)
+                                .build())
+                        .build()))
+                .build();
+        ConjureDefinitionValidator.validateAll(conjureDef, SafetyDeclarationRequirements.REQUIRED);
```

Comment:
Isn’t this unsafe? The external type is declared do-not-log, but it’s later declared as unsafe instead when a field is defined.

I don’t think we want to allow type safety from external type imports to be overridden when the type is used in a field.